lib/Interpreter/ValuePrinter.cpp

```diff
@@ -581,56 +580,44 @@ namespace cling {
   }
 } // end namespace cling
 
+namespace {
 
-template<typename T, bool> struct ExecutePrintValue;
-
-template<typename T>
-struct ExecutePrintValue<T, false> {
-  std::string operator()(const Value &V, const T &val) {
-    Interpreter *Interp = V.getInterpreter();
-    Value printValueV;
+static std::string callPrintValue(const Value& V, const void* Val) {
+  Interpreter *Interp = V.getInterpreter();
+  Value printValueV;
 
-    {
-      // Use an llvm::raw_ostream to prepend '0x' in front of the pointer value.
+  {
+    // Use an llvm::raw_ostream to prepend '0x' in front of the pointer value.
 
-      cling::ostrstream Strm;
-      Strm << "cling::printValue(";
-      Strm << getTypeString(V);
-      Strm << (const void*) &val;
-      Strm << ");";
+    cling::ostrstream Strm;
+    Strm << "cling::printValue(";
+    Strm << getTypeString(V);
+    Strm << &Val;
```

Comment:
It's as it was before, just the (const void*) cast removed.
I believe it is what is wanted as it's being dereferenced in the `getTypeString` call.
A lot of tests would fail if this wasn't correct.